contracts/examples/adder/proxies_struct_interactor_main.rs

```diff
@@ -1,60 +1,90 @@
-#![allow(clippy::too_many_arguments)]
-#![allow(clippy::type_complexity)]
 multiversx_sc::imports!();
 
-pub struct TxProxy<A>
+pub struct TxProxy;
+
+impl<Env> TxProxyTrait<Env> for TxProxy
 where
-    A: multiversx_sc::api::VMApi + 'static,
+    Env: TxEnv,
 {
-    pub address: ManagedOption<A, ManagedAddress<A>>,
+    type TxProxyMethods = TxProxyMethods<Env>;
+
+    fn env(self, env: Env) -> Self::TxProxyMethods {
+        TxProxyMethods { env }
+    }
 }
 
-impl<A> TxProxy<A>
-where
-    A: multiversx_sc::api::VMApi + 'static,
-{	fn init<
-		Arg0: multiversx_sc::codec::CodecInto<BigUint<A>>,
+impl<Env: TxEnv + multiversx_sc::api::CallTypeApi> TxProxyMethods<Env> {
+	pub fn init<
+		Arg0: multiversx_sc::codec::CodecInto<BigUint<Env>>,
 	>(
 		&mut self,
 		initial_value: Arg0,
-	) -> ContractDeploy<A, ()> {
-		let ___opt_address___ = multiversx_sc::extract_opt_address!(self);
-		let mut ___contract_deploy___ = multiversx_sc::constructors_proxy!(___opt_address___);
-		___contract_deploy___.push_endpoint_arg(&initial_value);
-		___contract_deploy___
+	) -> multiversx_sc::types::Tx<Env,
+        (),
+        (),
+        (),
+        (),
+        FunctionCall<<Env as multiversx_sc::types::TxEnv>::Api>,
```

Comment:
`init` produces a `DeployCall` here. 

Also, I'm pretty sure that in this case we can write `<Env::Api>` directly, hygiene should not be an issue, and it looks easier to read.